src/core/mac/mac.cpp

```diff
@@ -1028,6 +1028,8 @@ void Mac::BeginTransmit(void)
         OT_ASSERT(false);
     }
 
+    frame->SetTxPower(kRadioPowerInvalid);
```

Comment:
Suggest moving this before the `switch()` (and maybe setting it on all `txFrames` so it can be later used in multi-radio). This way the next layer methods which prepare the frame (e.g. `MeshForwarder:: HandleFrameRequest()` will have the option to change this on any specific frame if they want to.

Reply:
agreed. Moved it into `TxFrames::Clear()`.